### PR TITLE
using Regexp Match 

### PR DESCRIPTION
**Pull request by stewart-yu**
**What this PR does / why we need it**:
using regexp match achieve find efficiently

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/aws/instances.go [hunk 1]
```diff
@@ -76,8 +77,8 @@ func (name kubernetesInstanceID) mapToAWSInstanceID() (awsInstanceID, error) {
 
 	// We sanity check the resulting volume; the two known formats are
 	// i-12345678 and i-12345678abcdef01
-	// TODO: Regex match?
-	if awsID == "" || strings.Contains(awsID, "/") || !strings.HasPrefix(awsID, "i-") {
+	reg := regexp.MustCompile("^i-[^/]*$")
```
Comment (rphillips) [inline pkg/cloudprovider/providers/aws/instances.go]: the resulting `reg` could be made into a global variable, so it doesn't need to be compiled on every pass.

File: pkg/cloudprovider/providers/aws/volumes.go [hunk 2]
```diff
@@ -75,8 +76,8 @@ func (name KubernetesVolumeID) mapToAWSVolumeID() (awsVolumeID, error) {
 
 	// We sanity check the resulting volume; the two known formats are
 	// vol-12345678 and vol-12345678abcdef01
-	// TODO: Regex match?
-	if strings.Contains(awsID, "/") || !strings.HasPrefix(awsID, "vol-") {
+	reg := regexp.MustCompile("^vol-[^/]*$")
```
Comment (rphillips) [inline pkg/cloudprovider/providers/aws/volumes.go]: ditto here

File: pkg/cloudprovider/providers/aws/instances.go (hunk 1 above)
Comment (stewart-yu) [reply pkg/cloudprovider/providers/aws/instances.go]: I have thought about it , but get grid of it after consider readability of code. I will fix it, Tahnks @rphillips

Comment (stewart-yu) [reply pkg/cloudprovider/providers/aws/instances.go]: Bad news, the `regexp.MustCompile("^vol-[^/]*$")` can't initializer as const. @rphillips

**Comment by stewart-yu** [comment]
This PR need you approve again, thanks @jsafrane @rphillips

File: pkg/cloudprovider/providers/aws/volumes.go [hunk 3]
```diff
@@ -19,11 +19,15 @@ package aws
...
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 )
 
+// awsVolumeRegMatch represents Regex Match for AWS volume.
+var awsVolumeRegMatch *regexp.Regexp = nil
```
Comment (rphillips) [inline pkg/cloudprovider/providers/aws/volumes.go]: you would initialize these on line 29.

```go
var awsVolumeRegMatch = regexp.MustCompile("^vol-[^/]*$")
```

Comment (stewart-yu) [reply pkg/cloudprovider/providers/aws/volumes.go]: Fix it, thanks.

**Comment by stewart-yu** [comment]
cc @zmerlynn

**Pull request merged**